server/services/generateFixture.js

```diff
@@ -0,0 +1,63 @@
+class TeamNode {
+    // Make a Data structure for a team where they have teamId and adjacency list
+    constructor(teamId) {
+        this.teamId = teamId
+        this.played = []
+    }
+    // Add team to played/adjacency list
+    playTeam(teamNode) {
+        const found = this.played.find((playedTeam) => playedTeam.team.teamId === teamNode.teamId)
+        if (found) found.times += 1
+        else this.played.push({ team: teamNode, times: 1 })
+    }
+    // Find how many games this team has played
+    gamesPlayed() {
+        return this.played
+            .map((playedTeam) => playedTeam.times)
+            .reduce((prev, curr) => prev + curr, 0)
+    }
+    // Private method to find next team to play from adjacency list based on least amount of times played
+    #nextTeamFromPlayedTeams() {
+        return this.played.sort((t1, t2) => t2.times - t1.times)[0].team
```

Comment:
i might be wrong, but won't this give the team played with the highest `times` instead of lowest?

```js
> var played = [{team:'a',times:100},{team:'b',times:5}]
> played.sort((t1,t2) => t2.times - t1.times)
[ { team: 'a', times: 100 }, { team: 'b', times: 5 } ]
> played.sort((t1,t2) => t1.times - t2.times)
[ { team: 'b', times: 5 }, { team: 'a', times: 100 } ]
```

if you have tested it works as expected then ignore the suggestion

```suggestion
        return this.played.sort((t1, t2) => t1.times - t2.times)[0].team
```

Reply:
oops brain was lagging thanks for picking this up